src/main/java/picard/illumina/CollectIlluminaBasecallingMetrics.java

```diff
@@ -159,17 +174,19 @@ protected int doWork() {
                     barcodeToMetricCounts.put(barcode.toString(), new IlluminaMetricCounts(barcode.toString(), barcodeName, LANE));
                 }
             }
-
-            factory = barcodeToMetricCounts.isEmpty()
-                    ? new IlluminaDataProviderFactory(
+            if (USE_NEW_CONVERTER) {
+                factory = new IlluminaDataProviderFactory(BASECALLS_DIR, BARCODES_DIR, LANE, readStructure, bclQualityEvaluationStrategy);
```

Comment:
It looks to me that you could merge this `if` with the one below, since the only intervening line doesn't rely on `factory`. Once that's done, you can change the new converter line to: 
```
setupNewDataProvider(new IlluminaDataProviderFactory(BASECALLS_DIR, BARCODES_DIR, LANE, readStructure, bclQualityEvaluationStrategy));
```